dashboard/src/components/ErrorAlert/ErrorSelector.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { shallow } from "enzyme";
+import * as React from "react";
+
+import { PermissionsErrorAlert } from ".";
+import { definedNamespaces } from "../../shared/Namespace";
+import {
+  ConflictError,
+  ForbiddenError,
+  IRBACRole,
+  NotFoundError,
+  UnprocessableEntity,
+} from "../../shared/types";
+import ErrorPageHeader from "./ErrorAlertHeader";
+import ErrorSelector from "./ErrorSelector";
+import PermissionsListItem from "./PermissionsListItem";
+import UnexpectedErrorAlert from "./UnexpectedErrorAlert";
+
+describe("ConflictError", () => {
+  it("should render a simple message with the name of the resource", () => {
+    const wrapper = shallow(<ErrorSelector error={new ConflictError()} resource={"my app"} />);
+    const errAlert = wrapper.find(UnexpectedErrorAlert);
+    expect(errAlert).toExist();
+    expect(errAlert.html()).toContain("my app already exists, try a different name");
+    expect(errAlert.html()).not.toContain("Troubleshooting");
+    expect(wrapper).toMatchSnapshot();
+  });
+});
+
+describe("ForbiddenError", () => {
+  it("should show an error message with the default RBAC roles", () => {
+    const defaultRBACRoles = {
+      view: [
+        {
+          apiGroup: "v1",
+          namespace: "my-ns",
+          resource: "my-app",
+          verbs: ["get", "list"],
+        } as IRBACRole,
+      ],
+    };
+    const wrapper = shallow(
+      <ErrorSelector
+        error={new ForbiddenError()}
+        resource={"my app"}
+        namespace={"my-ns"}
+        defaultRequiredRBACRoles={defaultRBACRoles}
+        action="view"
+      />,
+    );
+    const errAlert = wrapper.find(PermissionsErrorAlert);
+    expect(errAlert).toExist();
+    expect(errAlert.html()).not.toContain("Troubleshooting");
+    const header = errAlert
+      .shallow()
+      .find(UnexpectedErrorAlert)
+      .shallow()
+      .find(ErrorPageHeader)
+      .shallow();
+    expect(header.text()).toContain(
+      "You don't have sufficient permissions to view my app in the my-ns namespace",
+    );
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should extract the required RBAC roles from the error message", () => {
+    const role = { apiGroup: "v1", namespace: "my-ns", resource: "my-app", verbs: ["get", "list"] };
+    const message = JSON.stringify([role]);
+    const wrapper = shallow(
+      <ErrorSelector error={new ForbiddenError(message)} resource="my-app" />,
+    );
+    const items = wrapper
```

Comment:
Wouldn't make more sense to create the wrapper using `mount` in these cases so you do not need to chase the items down the line?